test/mocked/raidenInspector.test.ts

```diff
@@ -0,0 +1,236 @@
+import mockito from "ts-mockito";
+import { expect, assert } from "chai";
+import { ethers } from "ethers";
+import { RaidenInspector, RaidenAppointment, RaidenTools } from "../../src/integrations/raiden";
+import { ChannelType } from "../../src/dataEntities";
+
+const isRejected = async (result: Promise<any>) => {
+    return await result.then(
+        () => {
+            assert.fail();
+        },
+        reject => {
+            expect(reject).to.exist;
+        }
+    );
+};
+
+describe("Raiden inspector", () => {
+    let provider;
+    const minimumDisputePeriod = 10;
+    const appointmentObj = {
+        expiryPeriod: 10000,
+        // cast here to inform the type guard on new RaidenAppointment
+        type: ChannelType.Raiden as ChannelType.Raiden,
+        stateUpdate: {
+            additional_hash: "0xbea4d9b77acbfb279382b573824ec01ebc6b2185ccd0db2d967ba600250bf9f4",
+            balance_hash: "0xe73f895f3e0439ede072b30229eac682208f45c1e1d939f2e8d3a71980778765",
+            channel_identifier: 1,
+            closing_participant: "0xdddEC4D561eE68F37855fa3245Cb878b10Eb1fA0",
+            closing_signature:
+                "0x078d3465de0781b3011323df87e55491124fecdb74df8ed5857bdedf63f9a242680afc7311d0681b09108c01f3f72b9fcdffd803bced1df1fb944e219588286a1b",
+            non_closing_participant: "0xccca21b97b27DefC210f01A7e64119A784424D26",
+            non_closing_signature:
+                "0xd77e9816a50c897a36dd02380b4789a07adb58f7a9cf91cf8c36447539a6bb6e1d118b035886e317189610dd3cdc8adbaeea4a78e5391876ba606189c1aeadd81b",
+            nonce: 2,
+            chain_id: 3,
+            token_network_identifier: "0x34636d2289588BcA4e0fcF863857386F56b44bdC"
+        }
+    };
+
+    before(() => {
+        const onChainSettleBlockNumber = 500;
+        const onChainNonce = 1;
+        const onChainStatus = 1;
+        const onChainBlockNumber = 200;
+        const spiedEthers = mockito.spy(ethers);
+
+        // contract
+        const tokenNetworkAddress = "0x34636d2289588BcA4e0fcF863857386F56b44bdC";
+        const party1Address = "0xccca21b97b27DefC210f01A7e64119A784424D26";
+        const party2Address = "0xdddEC4D561eE68F37855fa3245Cb878b10Eb1fA0";
+        const otherTokenNetwork = "0xf0afbed24d88ce4cb12828984bb10d2f1ad0e185"
+        const mockedProvider = mockito.mock(ethers.providers.JsonRpcProvider);
+        mockito.when(mockedProvider.getCode(tokenNetworkAddress)).thenResolve(RaidenTools.ContractDeployedBytecode);
+        mockito.when(mockedProvider.getCode(otherTokenNetwork)).thenResolve(RaidenTools.ContractDeployedBytecode);
+        mockito.when(mockedProvider.getCode(party1Address)).thenResolve("0x");
+        mockito.when(mockedProvider.getCode(party2Address)).thenResolve(RaidenTools.ContractDeployedBytecode.slice(0, RaidenTools.ContractDeployedBytecode.length - 20));
```

Comment:
Add a comment to note that getCode is mocking incorrect behavior for party1Address and party2Address (easy to miss)

Reply:
Done